packages/schema-generator/src/generate_schema.test.ts

```diff
@@ -0,0 +1,60 @@
+import { describe, it, mock } from 'node:test';
+import { SchemaGenerator, parseDatabaseUrl } from './generate_schema.js';
+import assert from 'node:assert';
+import {
+  TypescriptDataSchemaGenerator,
+  TypescriptDataSchemaGeneratorConfig,
+} from '@aws-amplify/graphql-schema-generator';
+import fs from 'fs';
+
+const mockGenerateMethod =
+  mock.fn<(config: TypescriptDataSchemaGeneratorConfig) => Promise<string>>();
+mockGenerateMethod.mock.mockImplementation(async () => {
+  return 'TYPESCRIPT_DATA_SCHEMA';
+});
+TypescriptDataSchemaGenerator.generate = mockGenerateMethod;
+
+const mockFsWriteFileSync = mock.method(fs, 'writeFileSync');
+mockFsWriteFileSync.mock.mockImplementation(() => {});
+
+void describe('SchemaGenerator', () => {
+  void it('should generate schema', async () => {
+    const schemaGenerator = new SchemaGenerator();
+    await schemaGenerator.generate({
+      connectionString: 'mysql://user:password@localhost:3306/db',
+      out: 'schema.ts',
+    });
+    assert.strictEqual(mockGenerateMethod.mock.calls.length, 1);
+    // assert.strictEqual(mockFsWriteFileSync.mock.calls.length, 1);
+  });
+
+  void it('should parse database url correctly', async () => {
+    const dbConfig = parseDatabaseUrl(
+      'mysql://user:password@localhost:3306/db'
```

Comment:
I'd be a bit more comfortable with this test if the hostname were more complex than "localhost". At least throw some dashes in one of the tests :) 

We should also check for proper handling of things like usernames and especially passwords, where characters are far more likely to fall outside the [a-zA-Z0-9] space and need to be URL escaped

Reply:
Added tests with special characters in hostname and password.